[FIX] website: adjust `header_sales_three` small btn

Brieuc-brd opened the pull request:
Since the update of Bootstrap from 5.1 to 5.3 in commit https://github.com/odoo-dev/odoo/commit/058212e12b5079eba870bde9775fe98f27928935, the font-size of `btn-sm` wasn't taken into account because the font-size of `nav-link` had the priority. This created a layout inconsistency in `header_sales_three` template.

In addition, once the user started changing the font size of the navigation in the Web editor (by triggering the `header-font-size` value), the font size of the small buttons was removed and replaced by the regular font size.

This commit adapts and forces the font-size of small links in the `header_sales_three` template to be always proportional to header font size.


| Before | After |
|--------|--------|
| <img width="650" alt="Capture d’écran 2024-08-16 à 11 00 04" src="https://github.com/user-attachments/assets/a3883f97-9de1-4940-9c40-b1db6dda64dd"> | <img width="549" alt="Capture d’écran 2024-08-16 à 10 58 52" src="https://github.com/user-attachments/assets/6ed2c695-55de-42b3-91e3-e9ae6d77c017"> |






---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

robodoo commented:
[![Pull request status dashboard](https://mergebot.odoo.com/odoo/odoo/pull/176801.png)](https://mergebot.odoo.com/odoo/odoo/pull/176801)

Brieuc-brd commented:
@stefanorigano 👋 
Here's another small fix, what do you think ?

stefanorigano commented on addons/website/static/src/scss/website.scss:
```diff
@@ -1221,6 +1221,9 @@ header {
         @include o-add-gradient('menu-secondary-gradient');
     }
+    #wrapwrap .o_header_sales_three_small_links {
+        --nav-link-font-size: #{$font-size-sm};
```
`$font-size-sm` looks correct when the header uses the default font-size, but as soon as it's customised proportions will mismatch. Also, customizing the header font-size seems to force inner items to `inherit`,  invalidating your rule 🤷🏻‍♂️ 

![image](https://github.com/user-attachments/assets/6aa3a3b5-1ba0-4220-8f62-6707b1d81ab1)

Brieuc-brd replied:
Good catch, thanks ! 
I've forced the font-size to be always proportional to header font size, what do you think ?

As we discussed, I also rebased on `master`.

stefanorigano reviewed (changes requested):
Hello @Brieuc-brd .. seems a bit more complicated than that 😬... read my comment in the code.
Also, I'm a bit doubtful that this change could be considered a FIX 🤔.
It's definitely an improvement but, if you don't look at the class names, the current "broken" behaviour could be considered the expected one as well by an user perspective... as said, not sure...🤔

stefanorigano commented on addons/website/static/src/scss/website.scss:
```diff
@@ -1291,6 +1291,10 @@ header {
     }
+    #wrapwrap header .o_header_sales_three_small_links .btn {
+        // Force small links to be proportional to header font size
+        font-size: .75em !important;
```
You get `12px` with the default font-size, and that's already very small... but if you set the header below `16px` you can easily fall into "really unreadable territory" 😅

I briefly tested `14px` (it was the standard just a few years ago) and you get ~`10px` that's not good for a menu entry 🥲

I have 2 proposals 

1. Use Boostrap default `$small-font-size`(=>  `.875em`)
```suggestion
        font-size: $small-font-size !important;
```

 2. Provide a failsafe value
```suggestion
        font-size: MAX(12px, .75em) !important;
```
I'll let you choose ;)

Brieuc-brd replied:
I would choose the second solution, thanks !

stefanorigano reviewed (approved):
LGTM 👍

Brieuc-brd commented:
@odoo/rd-website 👋 another small improvement ready for tech. review, could you take a look at it ? :)
Thanks in advance ! 👍

qsm-odoo reviewed (changes requested):
@Brieuc-brd There needs to be more than this IMO:

- It seems to work in 17.2 but not in 17.4, I guess this was broken when we updated Bootstrap from 5.1 to 5.3 with https://github.com/odoo/odoo/commit/058212e12b5079eba870bde9775fe98f27928935 ? -> If that's confirmed, could you mention it in the commit message ?

- Since the issue is also in 17.4 (introduced in the deprecated 17.3), I would fix it over there (you can just do as you would in master, it's recent enough to not care about stable policy here). @stefanorigano had the opposite opinion and asked you to change but here is my perspective:
   - Someone who started in 17.3 won't receive your fix as it needs an XML update, so it won't be a problem for them (it was ugly but if that's what they liked, it will stay that way).
   - Someone who **starts** after in 17.4 would have your fix, that would be nice as it really looks like ugly right now IMO
   - Someone who started <= 17.2 and had a beautiful header... would have an ugly header if they upgrade to 17.4 now :scream: The sooner we merge this in 17.4, the sooner we prevent this from happening :pray: 

- `nav-link btn-sm` was probably wrong to use from the start, I would have this commit do more and remove that inconsistency (`btn-sm` should just be removed I guess?)

- Instead of what you did, should not you rather change `--nav-link-font-size` in that specific `navbar-nav` environment?

Thanks :pray:

Brieuc-brd commented:
Hey @qsm-odoo ! Thanks for your review 🙏 

> If that's confirmed, could you mention it in the commit message ?

Indeed, the issue seems to come from there. I updated the commit and PR descriptions.

> Since the issue is also in 17.4 (introduced in the deprecated 17.3), I would fix it over there 

Ok, I retargeted the PR 👍 

> btn-sm should just be removed I guess?

Indeed, this class seems not necessary anymore, I removed it.

> Instead of what you did, should not you rather change --nav-link-font-size in that specific navbar-nav environment?

That's what I did at first, but we realized with @stefanorigano that it creates an issue when the user starts changing the font size using the editor: there's a css rule that resets the button font size when the header font size is changed 😬  :

https://github.com/odoo/odoo/blob/master/addons/website/static/src/scss/website.scss#L384

![image](https://github.com/user-attachments/assets/270d4301-94a0-429f-bf61-f20df22d7ee0)

Brieuc-brd commented:
> But if you remove the btn class, won't that solve that issue too? :)

Indeed, but this class is defined in the main `user_dropdown` template : https://github.com/odoo/odoo/blob/saas-17.4/addons/portal/views/portal_templates.xml#L120

This means I'd have to go through all the calls to make sure it doesn't break everything.
Since this is a FIX (I need to update the PR comment and title 🤦 ), should this kind of change be done in this PR?

qsm-odoo reviewed (approved):
> Since this is a FIX, should this kind of change be done in this PR?

Let's make those in the forward-ported version :+1: You're right, even removing the btn-sm ones now was already dubious :wink: 

@robodoo r+
